Sources/COWMacros/COWExpansionFactory.swift

```diff
@@ -218,30 +223,196 @@ internal class COWExpansionFactory<Context: MacroExpansionContext> {
   
   // MARK: Utilities
   
-  private func createDerivedStorageTypeDecl() -> StructDeclSyntax {
-    let members = appliedStructValidVarDecls.map {
-      MemberBlockItemSyntax(decl: $0)
+  // If `appliedStructDecl` conforms to Equatable and manually implements `==`,
+  // the derived storage must also conform to Equatable (otherwise we hit a
+  // compiler bug that breaks the build).
+  // https://forums.swift.org/t/macro-multiple-matching-functions-xyz-error-though-the-expanded-code-has-no-duplicate-functions-and-compiles/68700/4
+  // FIXME: remove the workaround when the compiler bug is fixed.
+  private func applyEquatableWorkaroundForDerivedStorage(
```

Comment:
`applyEquatableWorkaroundIfNeeded ` could be a better name. "IfNeeded" indicates that the contents added by this function are optional. This suffix improves the readability by hitting the reader to omit the details in this function if they are just trying to understand the outline of the program.